data_resource/storage/aws_s3.py

```diff
@@ -0,0 +1,91 @@
+import os
+import logging
+import boto3
+import json
+from botocore.exceptions import ClientError
+
+
+class S3Manager:
+    """S3Manager Manager Class."""
+
+    def __init__(self, config):
+        self.config = config
+        self.aws_access_key_id = self.config.AWS_ACCESS_KEY_ID
+        self.aws_secret_access_key = self.config.AWS_SECRET_ACCESS_KEY
+        self.region_name = self.config.AWS_S3_REGION
+
+        if (
+            self.aws_access_key_id == None
+            or self.aws_secret_access_key == None
+            or self.region_name == None
+        ):
+            raise RuntimeError("Invalid AWS S3 Configuration")
+
+    def required_env_check(self):
```

Comment:
If these or others are private methods prepend them with an underscore,

`def required_env_check(self):`
to
`def _required_env_check(self):`

Reply:
Made private, might need to be public in future.